reporter/jira.go

```diff
@@ -0,0 +1,196 @@
+			username = flags.String("jira.username", "JIRA_USERNAME", "", "User to authenticate to jira as")
+			password = flags.String("jira.password", "JIRA_PASSWORD", "", "Password for the user to authenticate")
+			fieldsConfiguration = flags.String("jira.fields", "JIRA_FIELDS", "", "all required fields seen in the online form in key:value format seperated by ; for multiple fields. Example - Project:ENG;Issue Type:Bug;Priority:Medium")
+			closedStatus = flags.String("jira.issue_closed_status", "JIRA_ISSUE_CLOSED_STATUS", "", "The status of issue when it is considered closed. This is used to decide whether or not to create new ticket for the same job")
+		},
+
+		Make: func() (Reporter, error) {
+			return newJiraReporter(*jiraURL, *username, *password, *fieldsConfiguration, *project, *issueType, *closedStatus)
+		},
+	})
+
+}
+
+func newJiraReporter(jiraURL, username, password, fieldsConfiguration, project, issueType, closedStatus string) (*jiraReporter, error) {
+
+	err := checkArgsNotNil(jiraURL, username, password, fieldsConfiguration, project, issueType, closedStatus)
+	if err != nil {
+		return nil, err
+	}
+	reporter := new(jiraReporter)
+	jiraClient, err := jira.NewClient(nil, jiraURL)
+	if err != nil {
+		return nil, fmt.Errorf("Could not create client: %s", err)
+
+	}
+
+	res, err := jiraClient.Authentication.AcquireSessionCookie(username, password)
+	if err != nil || !res {
+		return nil, fmt.Errorf("Authentication failed: %s", err)
+	}
+
+	if !jiraClient.Authentication.Authenticated() {
+		return nil, fmt.Errorf("Not authenticated")
+	}
+
+	reporter.client = jiraClient
+
+	err = reporter.setFieldsConfig(fieldsConfiguration)
+	if err != nil {
+		return nil, err
+	}
+
+	reporter.createdTasks = make(map[string]string)
+
+	reporter.closedStatusName = closedStatus
+
+	// get create meta information
+	meta, _, err := jiraClient.Issue.GetCreateMeta(project)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get create meta : %s", err)
+	}
+
+	// get right project
+	metaProject := meta.GetProjectWithKey(project)
+	if metaProject == nil {
+		return nil, fmt.Errorf("Could not find project with key %s", project)
+	}
+	reporter.metaProject = metaProject
+	// get right issue within project
+	metaIssuetype := metaProject.GetIssueTypeWithName(issueType)
+	if metaIssuetype == nil {
+		return nil, fmt.Errorf("Could not find issuetype %s. Available are %#v", issueType, getAllIssueTypeNames(metaProject))
+	}
+
+	// check if the given fields completes the mandatory fields and all listed fields are available
+	complete, err := metaIssuetype.CheckCompleteAndAvailable(reporter.fieldsConfig)
+	if !complete {
+		return nil, err
+	}
+
+	reporter.metaIssuetype = metaIssuetype
+	return reporter, nil
+}
+
+func (j *jiraReporter) Report(failure complainer.Failure, config ConfigProvider, stdoutURL, stderrURL string) error {
+
+	issueID, okay := j.createdTasks[failure.Name]
```

Comment:
This will work, but you'll have to extract the summary template from `JIRA_FIELDS`. You can also put task name into a separate JIRA field and search by that field instead of Summary.

The latter will work even if you put extra things from labels into Summary fields.